packages/server/src/middleware/nuxt.js

```diff
@@ -7,7 +7,22 @@ import { getContext } from '@nuxt/utils'
 export default ({ options, nuxt, renderRoute, resources }) => async function nuxtMiddleware(req, res, next) {
   // Get context
   const context = getContext(req, res)
-  const url = decodeURI(req.url)
+  let url
+
+  try {
+    url = decodeURI(req.url)
+  } catch (err) {
+    /* istanbul ignore if */
+    if (context && context.redirected) {
```

Comment:
You probably don't need that block here as it'll be caught further down the road.